src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,119 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
+
+const validationUrl = (value: string) => {
+  // eslint-disable-next-line max-len
+  const pattern = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+
+  return pattern.test(value);
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [movie, setMovie] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const getNewMovie = (name:string, newValue: string) => {
+    setMovie(state => ({ ...state, [name]: newValue }));
+  };
+
+  const checkRequired = (): boolean => {
+    const keys = Object.keys(movie);
+
+    return keys.filter(key => key !== 'description')
+      .some(key => movie[key as keyof Movie].trim() === '');
```

Comment:
try to avoid `as` in TS
```suggestion
      .some(key => movie[key].trim() === '');
```